plugins/inputs/mysql/mysql.go

```diff
@@ -920,74 +861,210 @@ func (m *Mysql) GatherUserStatisticsStatuses(db *sql.DB, serv string, acc telegr
 		return err
 	}
 	defer rows.Close()
-	var (
-		user                   string
-		total_connections      int64
-		concurrent_connections int64
-		connected_time         int64
-		busy_time              int64
-		cpu_time               int64
-		bytes_received         int64
-		bytes_sent             int64
-		binlog_bytes_written   int64
-		rows_fetched           int64
-		rows_updated           int64
-		table_rows_read        int64
-		select_commands        int64
-		update_commands        int64
-		other_commands         int64
-		commit_transactions    int64
-		rollback_transactions  int64
-		denied_connections     int64
-		lost_connections       int64
-		access_denied          int64
-		empty_queries          int64
-		total_ssl_connections  int64
-		count                  uint32
-	)
+
+	cols, err := convertColumns(rows.Columns())
+	if err != nil {
+		return err
+	}
+
+	read, err := getColSlice(len(cols))
+	if err != nil {
+		return err
+	}
+
+	if len(read) != len(cols) {
+		return fmt.Errorf("Results (%d) don't match up with columns (%d)!", len(read), len(cols))
+	}
 
 	servtag := getDSNTag(serv)
 	for rows.Next() {
-		err = rows.Scan(&user, &total_connections, &concurrent_connections,
-			&connected_time, &busy_time, &cpu_time, &bytes_received, &bytes_sent, &binlog_bytes_written,
-			&rows_fetched, &rows_updated, &table_rows_read, &select_commands, &update_commands, &other_commands,
-			&commit_transactions, &rollback_transactions, &denied_connections, &lost_connections, &access_denied,
-			&empty_queries, &total_ssl_connections, &count,
-		)
+		err = rows.Scan(read...)
 		if err != nil {
 			return err
 		}
 
-		tags := map[string]string{"server": servtag, "user": user}
-		fields := map[string]interface{}{
+		tags := map[string]string{"server": servtag, "user": *read[0].(*string)}
+		fields := map[string]interface{}{}
 
-			"total_connections":      total_connections,
-			"concurrent_connections": concurrent_connections,
-			"connected_time":         connected_time,
-			"busy_time":              busy_time,
-			"cpu_time":               cpu_time,
-			"bytes_received":         bytes_received,
-			"bytes_sent":             bytes_sent,
-			"binlog_bytes_written":   binlog_bytes_written,
-			"rows_fetched":           rows_fetched,
-			"rows_updated":           rows_updated,
-			"table_rows_read":        table_rows_read,
-			"select_commands":        select_commands,
-			"update_commands":        update_commands,
-			"other_commands":         other_commands,
-			"commit_transactions":    commit_transactions,
-			"rollback_transactions":  rollback_transactions,
-			"denied_connections":     denied_connections,
-			"lost_connections":       lost_connections,
-			"access_denied":          access_denied,
-			"empty_queries":          empty_queries,
-			"total_ssl_connections":  total_ssl_connections,
+		for i := range cols {
```

Comment:
I would use `for _, column := range cols {` and avoid using the index directly.  I know in some situations it can be slightly slower to iterate in this manner, but I think it almost always is more readable and less error prone.

Reply:
In this case, I think leaving it is best. I'll still need the iterator for the `read` slice and it's pretty clear where `cols[i]` is only used in the switch block.